timmyichen/dev-directory
# Cypress tests setup

Closes #15 (replace with issue number)
Description

Please include a summary of the changes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added a test based on cypress for the app being online with a rendered login button.

-     Created a config file in root for cypress
-     Placed cypress tests and related files in ./client/tests/cypress
-     Added a script, cypress:open to open cypress.
-     Added a subsection under the testing section in the readme on running cypress tests

Motivation: To implement front end testing with cypress.

Dependencies: Can be found at [the cypress installation documentation](https://docs.cypress.io/guides/getting-started/installing-cypress)

Testing

Describe what a reviewer should do to test your code, including instructions for manually testing as well as relevant automated test suites to run.

- install dependencies
- start the app
- run cypress:open
- go to e2e from the menu
- run the script cypress-base-test

Type of change

    New feature (non-breaking change which adds functionality)

Screenshots

NA

Checklist:

[x] Changes have new/updated automated tests, if applicable
[x] Changes have new/updated docs, if applicable
[x] I have performed a self-review of my own code
[x] I have added comments on any new, hard-to-understand code
[x] Changes have been manually tested
[x] Changes generate no new errors or warnings
